linera-core/src/local_node.rs

```diff
@@ -685,19 +675,22 @@ where
     async fn try_download_blob_from<A>(
         name: ValidatorName,
         node: &mut A,
-        chain_id: ChainId,
         blob_id: BlobId,
     ) -> Option<HashedBlob>
     where
         A: LocalValidatorNode + Clone + 'static,
     {
-        let query = ChainInfoQuery::new(chain_id).with_blob(blob_id);
-        if let Ok(response) = node.handle_chain_info_query(query).await {
-            if response.check(name).is_ok() {
-                return response.info.requested_blob;
+        match node.download_blob(blob_id).await.map(Blob::into_hashed) {
+            Ok(hashed_blob) if hashed_blob.id() == blob_id => Some(hashed_blob),
+            Ok(_) => {
+                tracing::info!("Validator {name} sent an invalid blob {blob_id}.");
+                None
+            }
+            Err(error) => {
+                tracing::debug!("Validator {name} failed to send blob {blob_id}: {error}");
```

Comment:
Optional nit: should this be worded like the error was in the client?

```suggestion
                tracing::debug!("Failed to fetch blob {blob_id} from validator {name}: {error}");
```